### PR TITLE
Massive UI improvement

### PR DESCRIPTION
**Pull request by dvvrd**
Docks behavior made rather more smart: now different sets of them is shown when different tabs are selected.
Two modes are introduced: editing and debugging. Edit mode is same as earlier, in the debugging one most of the window space is occupied by 2D model. It is now docked into interface, may be undocked to a separate window.

### CODE REVIEW

File: plugins/robots/common/twoDModel/include/twoDModel/engine/view/d2ModelWidget.h [hunk 1]
```diff
@@ -61,7 +61,7 @@ class TWO_D_MODEL_EXPORT D2ModelWidget : public utils::QRealDialog, public kitBa
 	D2ModelWidget(model::Model &model, QWidget *parent = 0);
 	~D2ModelWidget();
 
-	void init();
+	/// Overrides default closing behavoiur with just hiding window.
```
Comment (yurii-litvinov) [inline plugins/robots/common/twoDModel/include/twoDModel/engine/view/d2ModelWidget.h]: behavoiur -> behaviour

File: plugins/robots/interpreters/interpreterCore/include/interpreterCore/managers/actionsManager.h [hunk 2]
```diff
@@ -67,6 +67,15 @@ class ActionsManager : public QObject
 	/// change world in 2d model or configuration of robot sensors.
 	QAction &exportExerciseAction();
 
+	/// Action that changes current UI mode to debug: hides palette, property editor and so on,
+	/// opens variable values dock and maybe 2D model dock.
+	QAction &debugModeAction();
+
+	/// Action that changes current UI mode to edit: show palette and property editor if we are
+	/// editing diagram or variables list if we are editing the code.
+	/// Closess variable values dock and maybe 2D model dock.
```
Comment (yurii-litvinov) [inline plugins/robots/interpreters/interpreterCore/include/interpreterCore/managers/actionsManager.h]: Closes

File: plugins/robots/interpreters/interpreterCore/src/managers/uiManager.cpp [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+#include <qrutils/inFile.h>
+#include <qrutils/smartDock.h>
+
+#include "src/ui/modeStripe.h"
+
+using namespace interpreterCore;
+
+static const QColor backgrondColor = QPalette().color(QPalette::Background);
+static const QColor editModeColor = QPalette().color(QPalette::Background);
+//static const QColor editModeColor = qRgb(135, 206, 250);
```
Comment (yurii-litvinov) [inline plugins/robots/interpreters/interpreterCore/src/managers/uiManager.cpp]: мм?

File: plugins/robots/interpreters/interpreterCore/src/managers/uiManager.cpp [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+	case Mode::Editing:
+		result = new ui::ModeStripe(action, tr("Edit mode"), statusBar);
+		break;
+	case Mode::Debugging:
+		result = new ui::ModeStripe(action, tr("Debug mode"), statusBar);
+		break;
+	}
+
+	if (!result) {
+		qWarning() << "Forgot to implement producing status bar button for mode" << static_cast<int>(mode);
```
Comment (yurii-litvinov) [inline plugins/robots/interpreters/interpreterCore/src/managers/uiManager.cpp]: может, это в лог?

**Comment by yurii-litvinov** [comment]
по умолчанию в режиме отладки не видно окна с переменными

**Comment by yurii-litvinov** [comment]
и кнопочка "задокать" страшновата

**Comment by yurii-litvinov** [comment]
и графиков сенсоров по умолчанию тоже нет

**Comment by dvvrd** [comment]
все поправил, надо проверить

**Pull request merged**